src/Middleware/AmqpFlushMiddlware.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Jwage\PhpAmqpLibMessengerBundle\Middleware;
+
+use Jwage\PhpAmqpLibMessengerBundle\Stamp\Deferred;
+use Jwage\PhpAmqpLibMessengerBundle\Stamp\Flush;
+use Jwage\PhpAmqpLibMessengerBundle\Transport\BatchTransportInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+use Symfony\Component\Messenger\Stamp\SentStamp;
+use Symfony\Component\Messenger\Transport\Sender\SendersLocatorInterface;
+
+final class AmqpFlushMiddlware implements MiddlewareInterface
+{
+    /** @var array<BatchTransportInterface> */
+    private array $transportsToFlush = [];
+
+    public function __construct(
+        private SendersLocatorInterface $sendersLocator,
+    ) {
+    }
+
+    public function handle(Envelope $envelope, StackInterface $stack): Envelope
+    {
+        if ($envelope->getMessage() instanceof Flush) {
+            foreach ($this->transportsToFlush as $envelope) {
+                foreach ($this->sendersLocator->getSenders($envelope) as $transport) {
```

Comment:
this is a bit annoying - Symfony doesn't offer a way to get the transport by name, so instead we always keep the last envelope sent on a particular envelope ID then use the `SendersLocator` to find the transports/senders. It's not as super-efficicent as it could be but at the same time it's not the hotpath...